python_modules/dagster/dagster/_core/storage/runs/sql_run_storage.py

```diff
@@ -260,7 +260,13 @@ def _add_filters_to_query(self, query: SqlAlchemyQuery, filters: RunsFilter) ->
         check.inst_param(filters, "filters", RunsFilter)
 
         if filters.run_ids:
-            query = query.where(RunsTable.c.run_id.in_(filters.run_ids))
+            conditions = []
+            for run_id in filters.run_ids:
+                if len(run_id) < 36:
+                    conditions.append(RunsTable.c.run_id.like(f"{run_id}%"))
+                else:
+                    conditions.append(RunsTable.c.run_id == run_id)
+            query = query.where(db.or_(*conditions))
```

Comment:
Nice this is a great solution!

Does this need to be applied to other RunStorage implementations? (slash is there a separate one in cloud? not sure..)